index.js

```diff
@@ -0,0 +1,18 @@
+
+
+function calculateItemTotal(item) {
+  const subTotal = item.price * item.quantity * (1 - item.discount)
+
+  item.total = parseInt(subTotal * 100) / 100
```

Comment:
While map will return a new array making calculateTotal pure, this `item.total =` changes the object passed in to calculateItemTotal making it not pure.